src/components/CallBack/index.tsx

```diff
@@ -37,16 +41,24 @@ const CallBack: React.FC<RouteComponentProps> = (props) => {
         history.push('/');
     }
     useEffect(() => {
-        async function getUser() {
+        const getUser = async () =>  {
             try {
                 const response = await axios.post(
                     `${config.url}/users/github-init`,
                     { code },
                     { withCredentials: true }
                 );
+                let accessToken = null;
+                if(response.headers["set-authorization"]){
+                    //remove Bearer from the value
+                    accessToken =  response.headers["set-authorization"].substr(7);
+                }
+                const token: IToken = {
+                    accessToken
+                } 
+                dispatch(setAccessToken(token));
```

Comment:
We need to duplicate this code for the function (`onAccept` on L87 of this file) handling user registration as well. Otherwise a new user is unable to visit the home page since the token is never set for him.